internal/hcs/process.go

```diff
@@ -365,14 +381,19 @@ func (process *Process) Close() (err error) {
 		return nil
 	}
 
+	process.stdioLock.Lock()
+	defer process.stdioLock.Unlock()
```

Comment:
Since there's other stuff in this function, it's probably better to unlock just after the if blocks.

Reply:
Ditto elsewhere that we're not concerned about early returns...

Reply:
Curious how you would feel about a pattern like this:

```
	process.stdioLock.Lock()
	var unlockOnce sync.Once
	defer unlockOnce.Do(process.stdioLock.Unlock)
	if process.hasCachedStdio {
		in, out, err := process.stdin, process.stdout, process.stderr
		process.stdin, process.stdout, process.stderr = nil, nil, nil
		process.hasCachedStdio = false
		unlockOnce.Do(process.stdioLock.Unlock)
		return in, out, err, nil
	}
```

Too much overhead using a `Once` for such a simple thing? Or worth considering?

Reply:
No don't do that. :)